pkg/plugins/manager/fakes/fakes.go

```diff
@@ -204,6 +204,7 @@ type FakePluginRepo struct {
 	GetPluginArchiveFunc      func(_ context.Context, pluginID, version string, _ repo.CompatOpts) (*repo.PluginArchive, error)
 	GetPluginArchiveByURLFunc func(_ context.Context, archiveURL string, _ repo.CompatOpts) (*repo.PluginArchive, error)
 	GetPluginArchiveInfoFunc  func(_ context.Context, pluginID, version string, _ repo.CompatOpts) (*repo.PluginArchiveInfo, error)
+	PluginVersionFunc         func(pluginID, version string, compatOpts repo.CompatOpts) (repo.VersionData, error)
```

Comment:
Nit: Would be good to have `context.Context` here but you can save that a follow up if you prefer

Reply:
I prefer to make it in a follow up request. `PluginVersionFunc` is a mock of `PluginVersion` and PluginVersion is just a the old `pluginVersion` exported to reuse some code.